statsmodels/discrete/discrete_model.py

```diff
@@ -2631,27 +2634,27 @@ def _hessian_nb1(self, params):
 
         # for dl/dparams dalpha
         da1 = -alpha**-2
-        dldpda = np.sum(-mu/alpha * dparams + exog*mu/alpha *
-                        (-trigamma*mu/alpha**2 - 1/(alpha+1)), axis=0)
+        dldpda = np.sum(-a1 * dparams + exog * a1 *
+                        (-trigamma*mu/alpha**2 - prob), axis=0)
 
         hess_arr[-1,:-1] = dldpda
         hess_arr[:-1,-1] = dldpda
 
         # for dl/dalpha dalpha
-        digamma_part = (special.digamma(y + mu/alpha) -
-                        special.digamma(mu/alpha))
+        digamma_part = (special.digamma(y + a1) -
+                        special.digamma(a1))
 
-        log_alpha = np.log(1/(alpha+1))
+        log_alpha = np.log(prob)
         alpha3 = alpha**3
         alpha2 = alpha**2
         mu2 = mu**2
         dada = ((alpha3*mu*(2*log_alpha + 2*digamma_part + 3) -
-                2*alpha3*y + alpha2*mu2*trigamma +
+                2*alpha3*y +
                 4*alpha2*mu*(log_alpha + digamma_part) +
                 alpha2 * (2*mu - y) +
-                2*alpha*mu2*trigamma +
-                2*alpha*mu*(log_alpha + digamma_part) +
-                mu2*trigamma)/(alpha**4*(alpha2 + 2*alpha + 1)))
+                2*alpha*mu2*trigamma + mu2*trigamma + alpha2*mu2*trigamma +
+                2*alpha*mu*(log_alpha + digamma_part)
+                )/(alpha**4*(alpha2 + 2*alpha + 1)))
```

Comment:
This is a step of collecting-like-terms.  Put all of the trigamma terms in one place.  Next we'll put all of the `(log_alpha + digamma_part)` terms in one place.  Spoiler things are going to factor nicely.